xml/hwbp_registers.xml

```diff
@@ -434,6 +441,10 @@
             13: Matches when \FcsrMcontrolMatch$=5$ would not match.
 
             Other values are reserved for future use.
+
+            When \FcsrMcontrolSelect=1 and access size is N, a match only
+            compares the low N bits of the value to the low N bits of
+            \RcsrTdataTwo.
```

Comment:
I initially thought that this was fine but, to keep up what is undoubtedly my reputation as Mr. Corner-case: If match=4 then what is "the lower half of the value" for a load of 8b or 16b?  How about defining match=4 as "matches when the lower XLEN/2 bits of the value ANDed with the upper XLEN/2 bits of tdata2 match the lower XLEN/2 bits of tdata2"?  (Similar for match=5.)  Then it might be better for this sentence to say "... a match only compares the low N bits" (removing "of the value to the low N bits of tdata2" which implies that it's the low N bits of both things being compared, whatever those things may be).

I think that this sentence means that the newly-added sentence in select=1 is unnecessary because if you're not comparing those bits then it doesn't matter if you zero-extend or not.  Perhaps the new select=0 sentence should be removed and a sentence added down here like: "When select=0, a match only compares the low XLEN bits."

Reply:
I appreciate you're bringing up the corner cases. I wish they didn't exist, but that's largely a problem of my own making. (Except for XLEN being writable. I had nothing to do with that.)

This language about halfs is not precise enough, evidently. I'll make it more precise and more ugly.